### PR TITLE
DR2-1167 ingest folder opex creator

### PR DESCRIPTION
**Pull request by MancunianSam**
This creates an opex file for a folder. The folder id is passed as input
to the lambda.

### CODE REVIEW

File: src/test/scala/uk/gov/nationalarchives/LambdaTest.scala [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    val serveEvents = dynamoServer.getAllServeEvents.asScala
+    val queryEvent = serveEvents.head
+    val requestBody = queryEvent.getRequest.getBodyAsString
+    val expectedRequestBody =
+      """{"TableName":"test-table","IndexName":"test-gsi","KeyConditionExpression":"#A = :batchId AND #B = :parentPath",""" +
+        s""""ExpressionAttributeNames":{"#A":"batchId","#B":"parentPath"},"ExpressionAttributeValues":{":batchId":{"S":"TEST-ID"},":parentPath":{"S":"$folderParentPath/$folderId"}}}"""
+    expectedRequestBody should equal(requestBody)
+  }
+
+  "handleRequest" should "upload the opex file" in {
```
Comment (techncl) [inline src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:207]: This is only testing the path; do you think:

1. this title should be changed to "upload the opex file using the correct url" or something like that
2. it's worth adding a test to see if the correct opex is sent (either here or a brand new test)?

**Review by techncl — COMMENTED** [review]
Looks good, just one comment plus these 2:

1. I noticed that there are dynamo request checks but there are no checks for what we are sending to the s3 `.upload` and `.headObject` methods
2. For the XMLCreatorTest, do you think we should add a test for the if conditions, so:
    i. if title is blank
    ii. is not `hierarchyFolder` (i.e. title and name are the same)

**Comment by MancunianSam** [comment]
I've added in a test for checking the body uploaded to S3. The head object only gets the file size so that's checked in the same new test. 
I've added in another couple of tests for the XMLCreator as you've suggested.

File: src/test/scala/uk/gov/nationalarchives/LambdaTest.scala (hunk 1 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:207]: I've changed the title of this test and I've added another one in which checks what is sent to S3.

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**